src/main/rx/nrf24_cx10.c

```diff
@@ -0,0 +1,373 @@
+            while (NRF24L01_ReadReg(NRF24L01_17_FIFO_STATUS) & BV(NRF24L01_17_FIFO_STATUS_TX_FULL)) {
+                delayMicroseconds(100); // wait for room in the TX FIFO
+            }
+            XN297_WritePayload(payload, payloadSize);
+            NRF24L01_SetTxMode();// enter transmit mode to send the packet
+            delayMicroseconds(1000 / RF_CHANNEL_COUNT); // 1000us delay between each ACK
+        }
+        NRF24L01_SetRxMode();//reenter receive mode after sending ACKs
+        if (ackCount > ACK_TO_SEND_COUNT) {
+            NRF24L01_SetChannel(rfChannels[0]);
+            // and go into data state to wait for first data packet
+            protocolState = STATE_DATA;
+        }
+        break;
+    case STATE_DATA:
+        if (micros() > timeOfLastHop + hopTimeout) {
+            hopToNextChannel();
+            timeOfLastHop = micros();
+        }
+        // read the payload, processing of payload is deferred
+        if (NRF24L01_ReadPayloadIfAvailable(payload, payloadSize + 2)) {
+            XN297_UnscramblePayload(payload, payloadSize + 2);
+            hopToNextChannel();
+            ret = NRF24_RECEIVED_DATA;
+        }
+    }
+    return ret;
+}
+
+void cx10Init(const rxConfig_t *rxConfig, rxRuntimeConfig_t *rxRuntimeConfig)
+{
+    rxRuntimeConfig->channelCount = CX10_RC_CHANNEL_COUNT;
+    cx10Nrf24Init((nrf24_protocol_t)rxConfig->nrf24rx_protocol);
+}
+
+
+// XN297 emulation layer
+
+static int xn297_addrLen;
+static uint8_t  xn297_txAddr[5];
+static uint8_t  xn297_rxAddr[5];
+static uint8_t  xn297_crc = 1;
+static const uint8_t xn297_scramble[] = {
+    0xe3, 0xb1, 0x4b, 0xea, 0x85, 0xbc, 0xe5, 0x66,
+    0x0d, 0xae, 0x8c, 0x88, 0x12, 0x69, 0xee, 0x1f,
+    0xc7, 0x62, 0x97, 0xd5, 0x0b, 0x79, 0xca, 0xcc,
+    0x1b, 0x5d, 0x19, 0x10, 0x24, 0xd3, 0xdc, 0x3f,
+    0x8e, 0xc5, 0x2f};
+
+static const uint16_t xn297_crcXorout[] = {
+    0x0000, 0x3448, 0x9BA7, 0x8BBB, 0x85E1, 0x3E8C, // 1st entry is missing, probably never needed
+    0x451E, 0x18E6, 0x6B24, 0xE7AB, 0x3828, 0x8148, // it's used for 3-byte address w/ 0 byte payload only
+    0xD461, 0xF494, 0x2503, 0x691D, 0xFE8B, 0x9BA7,
+    0x8B17, 0x2920, 0x8B5F, 0x61B1, 0xD391, 0x7401,
+    0x2138, 0x129F, 0xB3A0, 0x2988};
+
+static uint8_t bitReverse(uint8_t bIn)
+{
+    uint8_t bOut = 0;
+    for (int i = 0; i < 8; ++i) {
+        bOut = (bOut << 1) | (bIn & 1);
+        bIn >>= 1;
+    }
+    return bOut;
+}
+
+static uint16_t crc16_update(uint16_t crc, unsigned char a)
+{
+    static const uint16_t crcPolynomial = 0x1021;
+    crc ^= a << 8;
+    for (int i = 0; i < 8; ++i) {
+        if (crc & 0x8000) {
+            crc = (crc << 1) ^ crcPolynomial;
+        } else {
+            crc = crc << 1;
+        }
+    }
+    return crc;
+}
+
+STATIC_UNIT_TESTED void XN297_SetTXAddr(const uint8_t* addr, int len)
+{
+    uint8_t buf[5] = { 0x55, 0x0F, 0x71, 0x0C, 0x00 }; // bytes for XN297 preamble 0xC710F55 (28 bit)
+    xn297_addrLen = len;
```

Comment:
xn297_addrLen is set here and then used in XN297_SetRXAddr

it will not work if functions are not called in correct order